### DISCUSSION TITLE
Change HTTP E2 extension to make HTTP requests clientside

### DISCUSSION THREAD

**Discussion by ghost** [wiremod/wire#2739] [Suggestions]
We already discussed that a lot in other issues, and I believe we all came to the conclusion we need to make that.

Pros:
- Increased security for servers
- HTTP extension can remain enabled by default without breaking any existing e2's

Cons:
- It will take a little longer to make a request because of 2 transfers added

If nobody is willing to make it, then I will, it just will take time cuz I don't have much of it.

**Comment by thegrb93** [comment]
I would do it if my data stream library was implemented. https://github.com/garrynewman/garrysmod/pull/941

That way transferring the data wouldn't require breaking it up manually to prevent channel overflows.

**Comment by AbigailBuccaneer** [comment]
@thegrb93 please can you [add me on Steam](http://steamcommunity.com/id/abigailbuccaneer/)? Thanks.

**Comment by thegrb93** [comment]
@AbigailBuccaneer done. I'm http://steamcommunity.com/profiles/76561197981780453/

**Comment by CaptainPRICE** [comment]
:+1:
